docs/commands.md

```diff
@@ -1,150 +1,538 @@
+The **RG.GETEXECUTION** command returns the execution [execution](functions.md#execution) details of a function that's in the executions list.
+
+**Redis API**
+
+```
+RG.GETEXECUTION <id> [SHARD|CLUSTER]
 ```
 
-Return an execution plan.
+_Arguments_
 
-Optional args:
-* `SHARD` - returns the local execution plan. This is the default for single-instance deployments.
-* `CLUSTER` - returns the execution plan from all shards in the cluster. This is the default for cluster deployments.
+* _id_: the [execution ID](functions.md#executionid) to get
+* _SHARD_: only gets the local execution (default in _stand-alone_ mode)
+* _CLUSTER_: collects all executions from shards (default in _cluster_ mode)
 
-Trivia: when called with the `CLUSTER` subcommand, RedisGears actually runs a gear that collects the shards' execution plans... self reference is so paradoxical :)
+!!! tip "RedisGears Trivia"
+    When called with the `CLUSTER` subcommand, RedisGears runs a RedisGears function that collects the shards' execution plans... self-reference is so paradoxical :)
 
-### Returns
-An array where each entry represents a shard. A shard's entry is structured to mimic a a Hash (i.e. alternating name and value entries) with the following data:
-* shard_id (string) - the shard's ID
-* execution_plan (Hash as array) - the exection plan
+_Return_
 
-Each shard's execution plan consists of the following:
-* status (string) - the execution plan's status
-* shards_received (integer) - the number of shards that received the plan from this shard
-* shards_completed (integer) - the number of shards that had completed plan received from this shard
-* results (integer) - the number of results collected (-1 if status is not done)
-* errors (array) - the errors reported by the steps (empty array if not done)
-* total_duration (integer) - the total duration of the plan in milliseconds
-* read_duration (integer) - the duration of the plan's read stage in milliseconds
-* steps (array) - the plan's steps
+An array if successful, or an error if the execution does not exist or is still running. The array consists of alternating key name and value entries as follows:
 
-A note about durations: the total duration is greater than the sum of step durations due to the execution's overheads.
+* **shard_id**: the shard's identifier in the cluster
+* **execution_plan**: the plan of execution is made of:
+    * **status**: the status (e.g. 'pending', 'running', 'done'?)
+    * **shards_received**: number of shards that received the execution
+    * **shards_completed**: number of shards that completed the execution
+    * **results**: count of results returned
+    * **errors**: count of errors raised
+    * **total_duration**: total execution duration in milliseconds
+    * **read_duration**: reader execution duration in milliseconds
+    * **steps**: the plan's steps are made of the following:
+        * **type**: step type
+        * **duration**: the step's duration in milliseconds
```

Comment:
Will be measured only when ProfileExecutions config val is 1 otherwise will show always zero..

Reply:
Resolved via 079f280